core/src/main/java/org/apache/accumulo/core/util/HostAndPort.java

```diff
@@ -281,4 +282,13 @@ public int getPortOrDefault(int defaultPort) {
     return hasPort() ? port : defaultPort;
   }
 
+  /**
+   * HostAndPort must implement compareTo. As this is a seldom used utiltiy, compareTo simply treats
+   * HostAndPort values as Strings.
+   */
+  @Override
+  public int compareTo(HostAndPort other) {
+    return Comparator.nullsFirst(Comparator.comparing(HostAndPort::toString)).compare(this, other);
```

Comment:
It doesn't look like the port value is padded, so using the toString value won't necessarily compare properly if the host is the same and the port differs in the number of digits (e.g., when using multiple tservers per host and os-selected tserver ports, this could happen depending on the OS configuration).

Reply:
I added a couple of new test cases to verify that the HostAndPort class handles padded port values.

Reply:
The case I'm thinking of is where a host, say localhost, has ports 9999 and 10000. The 10000 will sort first if you use a string comparison from toString, and second if you were actually comparing the integer port values.

Reply:
See the comment I made above regarding `thenComparingInt` for comparing the port.

Reply:
@brianloss modified the comparator to treat port values as ints and added test cases to verify.